airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/generate_erd/pipeline.py

```diff
@@ -0,0 +1,176 @@
+            .stdout()
+        )
+        return self._get_schema_from_discover_output(discover_output)
+
+    def _get_default_config(self) -> Secret:
+        if not self.context.local_secret_store:
+            raise ValueError("Expecting local secret store to be set up but couldn't find it")
+
+        filtered_secrets = [secret for secret in self.context.local_secret_store.get_all_secrets() if secret.file_name == "config.json"]
+        if not filtered_secrets:
+            raise ValueError("Expecting at least one secret to match name `config.json`")
+        elif len(filtered_secrets) > 1:
+            print(f"Expecting only one secret with name `config.json but got {len(filtered_secrets)}. Will take one randomly...")
+
+        return filtered_secrets[0]
+
+    @staticmethod
+    def _get_schema_from_discover_output(discover_output: str) -> str:
+        for line in discover_output.split("\n"):
+            json_line = json.loads(line)
+            if json_line.get("type") == "CATALOG":
+                return json.dumps(json.loads(line).get("catalog"))
+        raise ValueError("No catalog was found in output")
+
+    def _build_erd_container(self, connector_directory: Directory, discovered_catalog: str) -> Container:
+        """Create a container to run ERD generation."""
+        container = with_poetry(self.context)
+        if self.context.genai_api_key:
+            container = container.with_env_variable("GENAI_API_KEY", self.context.genai_api_key.value)
+
+        container = (
+            container.with_mounted_directory("/source", connector_directory)
+            .with_new_file("/source/erd/discovered_catalog.json", contents=discovered_catalog)
+            .with_mounted_directory("/app", self.context.erd_dir)
+            .with_workdir("/app")
+        )
+
+        return container.with_exec(["poetry", "lock", "--no-update"]).with_exec(["poetry", "install"])
+
+
+class UploadDbmlSchema(Step):
+    context: ConnectorContext
+
+    title = "Upload DBML file to dbdocs.io"
+
+    def __init__(self, context: PipelineContext) -> None:
+        super().__init__(context)
+
+    async def _run(self, connector_to_discover: Container) -> StepResult:
+        if not self.context.dbdocs_token:
+            raise ValueError(
+                "In order to publish to dbdocs, DBDOCS_TOKEN needs to be provided. Either pass the value or skip the publish step"
+            )
+
+        connector = self.context.connector
+        source_dbml_content = open(_get_dbml_file(connector.code_directory)).read()
+
+        dbdocs_container = await (
+            self.dagger_client.container()
+            .from_("node:lts-bullseye-slim")
+            .with_exec(["npm", "install", "-g", "dbdocs"])
+            .with_env_variable("DBDOCS_TOKEN", self.context.dbdocs_token.value)
+            .with_workdir("/airbyte_dbdocs")
+            .with_new_file("/airbyte_dbdocs/dbdocs.dbml", contents=source_dbml_content)
```

Comment:
nit: You could pass in the dbml dagger file from the container you ran the erd package in:

```suggestion
    async def _run(self, connector_to_discover: Container, dbml_file: File) -> StepResult:
        if not self.context.dbdocs_token:
            raise ValueError(
                "In order to publish to dbdocs, DBDOCS_TOKEN needs to be provided. Either pass the value or skip the publish step"
            )

        connector = self.context.connector

        dbdocs_container = await (
            self.dagger_client.container()
            .from_("node:lts-bullseye-slim")
            .with_exec(["npm", "install", "-g", "dbdocs"])
            .with_env_variable("DBDOCS_TOKEN", self.context.dbdocs_token.value)
            .with_workdir("/airbyte_dbdocs")
            .with_file("/airbyte_dbdocs/dbdocs.dbml", dbml_file)
```

This would avoid a back and forth to disk.
If you really want to use the file stored in our git repo I'd suggest using:
```
dbml_file = self.context.get_connector_dir(include=_get_dbml_file(connector.code_directory)).file(_get_dbml_file(connector.code_directory)))
```

I prefer to use dagger files as it slightly decouples the `airbyte-ci` code from the repo structure. File could come from the host or a container, which is pretty flexible.

Reply:
From the suggested change above, we could pass a `dbml_file: File` parameter to `_run`. I am wondering how to do that. Can we expect `GenerateDbml._run` to return an output? Can this output be `self.dagger_client.file(self._get_erd_folder(self.context.connector.code_directory))` and that would allow us to have access to `self._get_erd_folder(...) / "source.dbml"` in the `UploadDbmlSchema` step?

Reply:
You can pass an `output` parameter in the StepResult of GenerateDbml._run (which would be the file). Kinda   the same as what we do in the[ Build step.](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/build_image/steps/common.py#L66)